lib/editor/components/CreateSnapshotModal.js

```diff
@@ -13,50 +13,83 @@ import {connect} from 'react-redux'
 
 import * as snapshotActions from '../actions/snapshots.js'
 import {getComponentMessages} from '../../common/util/config'
+import { getRouteStatusText } from '../util/index.js'
 import {formatTimestamp} from '../../common/util/date-time'
-
-import type {Feed} from '../../types'
-import type {AppState} from '../../types/reducers'
+import type {Feed, GtfsRoute} from '../../types'
+import type {AppState, LockState} from '../../types/reducers'
+import {fetchRouteEntities} from '../actions/active'
 
 type Props = {
   createSnapshot: typeof snapshotActions.createSnapshot,
-  feedSource: Feed
-}
+  feedSource: Feed,
+  fetchRouteEntities: typeof fetchRouteEntities,
+  lock: LockState
+};
 
 type State = {
   comment: ?string,
+  confirmPublishWithUnapproved: Boolean,
+  loading: boolean,
   name: ?string,
   publishNewVersion: boolean,
-  showModal: boolean
+  showModal: boolean,
+  unapprovedRoutes: Array<GtfsRoute>
 }
 
 function getDefaultState () {
   return {
     comment: null,
     name: formatTimestamp(),
     publishNewVersion: false,
-    showModal: false
+    confirmPublishWithUnapproved: false,
+    showModal: false,
+    unapprovedRoutes: [],
+    loading: false
   }
 }
 
 class CreateSnapshotModal extends Component<Props, State> {
   state = getDefaultState()
 
   messages = getComponentMessages('CreateSnapshotModal')
+  onClose = null
 
   _onTogglePublish = (e: SyntheticInputEvent<HTMLInputElement>) => {
     this.setState({[e.target.name]: e.target.checked})
   }
 
+  _lockonTogglePublishUnapproved = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({[e.target.name]: e.target.checked})
+  }
+
   _onChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
     this.setState({[e.target.name]: e.target.value})
   }
 
   close = () => {
+    if (this.onClose) { this.onClose() }
     this.setState(getDefaultState())
   }
 
-  open () {
+  updateUnapprovedRoutes = () => {
+    const {fetchRouteEntities} = this.props
+    this.setState({loading: true})
+    // FixMe: type issue causing .then to have red underline.
+    fetchRouteEntities().then(data => {
```

Comment:
The tests are failing because the flow checks are failing because types here are wrong! The method doesn't return the data (since it does dispatch), which results in a race condition! 

A new dispatch action that moves handling all this state to redux is probably the cleanest way to fix it, but I will leave it up to second reviewer to decide the best way to handle this (could also try to refactor `fetchRouteEntites` to return correct type?)